src/Orleans/Streams/PersistentStreams/IStreamQueueCheckpoint.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams
+{
+    public interface IStreamQueueCheckpoint<TCheckPointState>
```

Comment:
is there a better name for this service so that it's not a noun? otherwise it kind of feels like a data holder. You'll know better than a non-native English speaker, but maybe `IStreamQueueCheckpointer`?